infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/Orchestrator.java

```diff
@@ -75,6 +81,30 @@ void handleQuery(CorfuPayloadMsg<OrchestratorRequest> msg, ChannelHandlerContext
                 .payloadMsg(new OrchestratorResponse(resp)));
     }
 
+    void removeNode(CorfuPayloadMsg<OrchestratorRequest> msg, ChannelHandlerContext ctx, IServerRouter r) {
+        RemoveNodeRequest req = (RemoveNodeRequest) msg.getPayload().getRequest();
+
+        if (findUUID(req.getEndpoint()) != null) {
+            // An add node workflow is already executing for this endpoint, return
+            // existing workflow id.
+            OrchestratorResponse resp = new OrchestratorResponse(
+                    new CreateWorkflowResponse(findUUID(req.getEndpoint())));
+            r.sendResponse(ctx, msg, CorfuMsgType.ORCHESTRATOR_RESPONSE
+                    .payloadMsg(resp));
+            return;
+        } else {
+            // Create a new workflow for this endpoint and return a new workflow id
+            Workflow workflow = getWorkflow(msg.getPayload());
```

Comment:
Is this best effort or is the driver single threaded? Because this is not protected against multiple threads?

Reply:
I will make it synchronized.

Reply:
Done in dispatch method.